codec-http2/src/main/java/io/netty/handler/codec/http2/Http2MultiplexCodec.java

```diff
@@ -399,6 +392,21 @@ public final void channelRead(ChannelHandlerContext ctx, Object msg) throws Exce
         super.channelRead(ctx, msg);
     }
 
+    @Override
+    public final void channelWritabilityChanged(final ChannelHandlerContext ctx) throws Exception {
+        forEachActiveStream(new Http2FrameStreamVisitor() {
```

Comment:
Note that this can easily starve later streams. What does the writability of the parent channel even mean to the child, since the scheduler is choosing how much to write? Another stream could have buffered 1 MB and the parent channel is writable only because we haven't flush()ed.

This seems very expensive as we don't generally need to loop through _all_ streams, and this will happen _every_ write cycle. That alone seems prohibitive.

Reply:
@ejona86 so you suggest to complete remove the "parent" relation ship here and also in `isWritable` ?

Reply:
I think removing the "parent" relationship is better than what is here. In my design I had expected to ignore the parent's writability.

But I could accept somewhere in-between if we can come up with something. Like: a stream only goes writable after the parent goes writable, but then it will stay writable even as the parent's writability changes, until the stream goes unwritable due to its own writes at which point it will wait for the parent again. That could (eventually) be implemented without looping over all streams and would have significantly fewer notifications. It would also prevent starvation (fairness could still be impacted, though). Other ideas welcome.

Reply:
@ejona86 I see... Let me think about if I can do this cheaply. If not you are happy with the rest (when the parent relationship would be completely removed) ?

Reply:
The rest looks fine.

It looks like the previous onHttp2StreamWritabilityChanged changed when the channel writability changed, so I guess the issue I'm raising here is pre-existing.